tests/misc/test_file_server.py

```diff
@@ -17,3 +17,37 @@ def test_fileserver_redirect_https(host):
     lines = stdout.decode().splitlines()
     assert lines[0].strip() == "HTTP/1.1 301 Moved Permanently"
     assert lines[2].strip() == "location:https://" + host.hostname_or_ip + path
+
+def test_fileserver_hsts(host):
+    def get_header():
+        process = subprocess.Popen(
+            ["curl", "-XGET", "-k", "-I", "https://" + host.hostname_or_ip],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        stdout, _ = process.communicate()
+        return stdout.decode().splitlines()
+
+    # By default HSTS header is not set by XAPI server.
+    for line in get_header():
+        assert line != "Strict-Transport-Security:max-age=63072000"
```

Comment:
`"Strict-Transport-Security:max-age=63072000"` could me put in variable since used twice.